### ISSUE TITLE
3.4 Block Exchange - BitSwap Protocol

### ISSUE THREAD

**Issue by mitar** [ipfs/papers#15]
In the current draft of the paper, it is written:

> In some cases, nodes must work for their blocks. In the case that a node has nothing that its peers want (or nothing at all), it seeks the pieces its peers want, with lower priority than what the node wants itself. This incentivizes nodes to cache and disseminate rare pieces, even if they are not interested in them directly.

When I read that I  got concerned about legal issues of just blindly and automatically downloading content from peers. In https://github.com/ipfs/faq/issues/47 I then read design goals and answer to "but bitswap says it may download stuff for others, to do better?" which is a good read.

I would suggest that both design goals and those questions and answers in some form are added to the paper. Design goals somewhere early on.

**Comment by alphaCTzo7G** [comment]
Agree.. There is some misinformation going around based on the paper, which might have significant PR (not pull requests :)) issues.. and may cause issues with adoption..

https://news.ycombinator.com/item?id=12809259

https://github.com/ipfs/go-ipfs/issues/4083#issuecomment-316561782
